include/alpaka/mem/buf/BufGenericSycl.hpp

```diff
@@ -147,68 +158,89 @@ namespace alpaka::trait
     {
         template<typename TExtent>
-        static auto allocBuf(DevGenericSycl<TPltf> const& dev, TExtent const& ext)
-            -> BufGenericSycl<TElem, TDim, TIdx, DevGenericSycl<TPltf>>
+        ALPAKA_FN_HOST static auto allocBuf(DevGenericSycl<TPltf> const& dev, TExtent const& extent)
+            -> BufGenericSycl<TElem, TDim, TIdx, TPltf>
         {
             ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
 
+#    if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
             if constexpr(TDim::value == 0 || TDim::value == 1)
             {
-                auto const width = getWidth(ext);
+                auto const width = getWidth(extent);
 
-#    if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
                 auto const widthBytes = width * static_cast<TIdx>(sizeof(TElem));
                 std::cout << __func__ << " ew: " << width << " ewb: " << widthBytes << '\n';
-#    endif
-
-                auto const range = sycl::range<1>{width};
-                return {dev, sycl::buffer<TElem, 1>{range}, ext};
             }
             else if constexpr(TDim::value == 2)
             {
-                auto const width = getWidth(ext);
-                auto const height = getHeight(ext);
+                auto const width = getWidth(extent);
+                auto const height = getHeight(extent);
 
-#    if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
                 auto const widthBytes = width * static_cast<TIdx>(sizeof(TElem));
                 std::cout << __func__ << " ew: " << width << " eh: " << height << " ewb: " << widthBytes
                           << " pitch: " << widthBytes << '\n';
-#    endif
-
-                auto const range = sycl::range<2>{width, height};
-                return {dev, sycl::buffer<TElem, 2>{range}, ext};
             }
             else if constexpr(TDim::value == 3)
             {
-                auto const width = getWidth(ext);
-                auto const height = getHeight(ext);
-                auto const depth = getDepth(ext);
+                auto const width = getWidth(extent);
+                auto const height = getHeight(extent);
+                auto const depth = getDepth(extent);
 
-#    if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
                 auto const widthBytes = width * static_cast<TIdx>(sizeof(TElem));
                 std::cout << __func__ << " ew: " << width << " eh: " << height << " ed: " << depth
                           << " ewb: " << widthBytes << " pitch: " << widthBytes << '\n';
+            }
 #    endif
 
-                auto const range = sycl::range<3>{width, height, depth};
-                return {dev, sycl::buffer<TElem, 3>{range}, ext};
-            }
+            TElem* memPtr = sycl::malloc_device<TElem>(
+                static_cast<std::size_t>(getExtentProduct(extent)),
+                dev.getNativeHandle().first,
+                dev.getNativeHandle().second);
+            auto deleter = [&dev](TElem* ptr) { sycl::free(ptr, dev.getNativeHandle().second); };
```

Comment:
Avoid redundant calls if possible:
```suggestion
            const auto& [nativeDev, nativeContext] = dev.getNativeHandle();
            TElem* memPtr = sycl::sycl <TElem>(
                static_cast<std::size_t>(getExtentProduct(extent)),
                nativeDev,
                nativeContext);
            auto deleter = [nativeContext](TElem* ptr) { sycl::free(ptr, nativeContext); };
```
(untested code, but something along these lines)
Also be careful about the lifetime of the captures from the deleter.